.github/workflows/multi-platform.yml

```diff
@@ -0,0 +1,238 @@
+name: Docker Publish Multi-Platform
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    # tags:
+    #   - 'v*'
+
+env:
+  REGISTRY: docker.io
+  # REGISTRY: ghcr.io # For GitHub Container Registry
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-amd64:
+    name: Build amd64 Docker Image
+    runs-on: ubuntu-latest
+    if: contains(github.event.head_commit.message, '[no build]') == false
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Prepare
+        run: |
+          platform=${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+          echo "IMAGE_NAME_LC=${IMAGE_NAME,,}" >> $GITHUB_ENV
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          # images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LC }} # Use this for GitHub Container Registry
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
+        with:
+          username: ${{ secrets.docker_hub_username }}
+          password: ${{ secrets.docker_hub_access_token }}
+          # If using GitHub Container Registry, comment out the above lines
+          # and uncomment the lines below
+          # registry: ${{ env.REGISTRY }}
+          # username: ${{ github.actor }}
+          # password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and push by digest
+        id: build
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ./images/${{ matrix.platform }}/Dockerfile
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          platforms: ${{ matrix.platform }}
+          push: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
The intention of lines like this is to allow docker builds on PRs but not pushes, since the original workflow also ran on PRs.

This workflow should behave similarly (run on PRs to verify build success but nothing more).

Reply:
One problem with that is we have to manually add the version number to meta step and update the version in workflow everytime. So, I use semver in release tags to automatically set it. 

But since these builds take significantly longer, I see your point. Let me add that to workflow.